p2p/swarm.go

```diff
@@ -422,12 +427,23 @@ func (s *swarm) onRemoteClientMessage(msg net.IncomingMessageEvent) error {
 		spew.Dump(pm)
 		panic("this is a defected message") // todo: Session bug, session scrambles messages and remove metadata
 	}
-	// authenticate message author - we already authenticated the sender via the shared session key secret
-	err = message.AuthAuthor(pm)
+
+	sign := pm.Metadata.AuthorSign
+	pm.Metadata.AuthorSign = nil
+
+	data, err := proto.Marshal(pm)
 	if err != nil {
+		e := fmt.Errorf("invalid msg format %v", err)
+		return e
+	}
+
+	if !session.VerifySignature(data, sign) {
 		return ErrAuthAuthor
 	}
 
+	pm.Metadata.AuthorSign = sign
```

Comment:
Should AuthorSign be an HMAC signature ? if this message will be passed (gossip) the receivers might not be able to verify it.